lib/OpenQA/Scheduler/Model/Jobs.pm

```diff
@@ -79,6 +80,7 @@ sub _allocate_jobs ($self, $free_workers) {
         my $tobescheduled = _to_be_scheduled($j, $scheduled_jobs);
         next if defined $allocated_jobs->{$j->{id}};
         next unless $tobescheduled;
+        OpenQA::Scheduler::Model::WorkerSlotPicker->new($tobescheduled)->pick_slots_with_common_worker_host;
```

Comment:
This should probably be a utility function instead of a model class, since it doesn't rerally represent anything stored longer term.